File: EjercicioLab1.c
#include <stdio.h>
#include <stdlib.h>
#define  ERR  500	

int RestaSucesiva (int a, int b);
int SumaSucesiva (int a, int b);
int menu ();

int main ()
{
	system("cls");
	
	int divisor, dividendo;
	
	printf("Ingresa el dividendo: ");
	scanf("%d", &dividendo);
	printf("Ingresa el divisor diferente a 0: ");
	scanf("%d", &divisor);
	
	RestaSucesiva(divisor, dividendo);
	
	return 0;
}

int menu ()
{
	int opc;
	
	do
	{
	printf("1. Divisi%cn por Restas Sucesivas.\n", 162);
	printf("2. Multiplicaci%on por Sumas Sucesivas.", 162);
	printf("3. Factorial de un n%cmero.\n", 163);
	printf("4. Salir.\n");
	
	scanf("%d", &opc);
	}
	while(opc=!4);
	
	
}

int RestaSucesiva (int divisor, int dividendo)
{
	int cociente = 0, residuo = dividendo;
	
	
	if (divisor == 0)
	{
		printf("\nError, no se puede dividir por cero\n");
		return ERR;
	}
	
	
	printf("\n\n%d dividido por %d ", dividendo, divisor);
	
	
	while (divisor<=dividendo)
	{
	  dividendo-=divisor;
	  cociente++;
	}
	residuo = dividendo;
	
	printf("nos da un cociente de %d y un residuo de %d\n", cociente, residuo);
	
	return 0;
}

int SumaSucesiva (int factor1, int factor2)
{
	
}

File: EjercicioLab1.py
def division_restas_sucesivas(dividendo, divisor):
    cociente = 0

    if divisor == 0:
        raise ValueError("No se puede dividir por cero")

    while dividendo >= divisor:
        dividendo -= divisor
        cociente += 1

    return cociente, dividendo
    
def multiplicacion_sumas_sucesivas(a, b):
    resultado = 0
    for _ in range(b):
        resultado += a
    return resultado
    
def factorial(n):
    if n < 0:
        raise ValueError("El factorial no está definido para números negativos")
    if n == 0:
        return 1
    resultado = 1
    for i in range(1, n + 1):
        resultado *= i
    return resultado

# Ejemplo de funcion 1:
dividendo = 10
divisor = 3
cociente, residuo = division_restas_sucesivas(dividendo, divisor)
print(f"{dividendo} dividido por {divisor} es igual a {cociente} con un residuo de {residuo}")

# Ejemplo de funcion 2:
numero1 = 5
numero2 = 3
resultado_multiplicacion = multiplicacion_sumas_sucesivas(numero1, numero2)
print(f"{numero1} multiplicado por {numero2} es igual a {resultado_multiplicacion}")



# Ejemplo de uso:
numero = 5
resultado_factorial = factorial(numero)
print(f"El factorial de {numero} es {resultado_factorial}")
